File: 0141-linked-list-cycle/0141-linked-list-cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {

        if(head==NULL){
            return false;
        }
        ListNode *wp = head;
        ListNode *nextP = head;
        while(nextP!=NULL && nextP->next!=NULL){

            wp = wp->next;
            nextP = nextP->next->next; 
            if(wp == nextP){
                return true;
            }
        }

        return false;
    }
};

File: 0141-linked-list-cycle/0141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {
        ListNode wp = head;
        ListNode nextP = head;

        while(nextP != null && nextP.next != null){
            wp = wp.next;
            nextP = nextP.next.next;

            if(wp == nextP){
                return true;
            }
        }
        return false;
    }
}